easybuild/tools/loose_version.py

```diff
@@ -0,0 +1,84 @@
+# This file contains the LooseVersion class based on the class with the same name
+# as present in Python 3.7.4 distutils.
+# The original class is licensed under the Python Software Foundation License Version 2.
+# It was slightly simplified as needed to make it shorter and easier to read.
+# In particular the following changes were made:
+# - Subclass object directly instead of abstract Version class
+# - Fully init the class in the constructor removing the parse method
+# - Always set self.vstring and self.version
+# - Shorten the comparison operators as the NotImplemented case doesn't apply anymore
+# - Changes to documentation and formatting
+
+import re
+# Modified: Make this compatible with Python 2
+try:
+    from itertools import zip_longest
+except ImportError:
+    # Python 2
+    from itertools import izip_longest as zip_longest
+
+
+class LooseVersion(object):
```

Comment:
Maybe we should rename this to something else than `LooseVersion`?

It's clearly based on that, but it's been fiddled with quite a bit, and behaves a bit different for some specific cases, so not naming it `LooseVersion` makes sense to me to avoid confusion with the original?

Reply:
I find the name good and would like to keep it:

- It really is a "loose version" as it doesn't care much about the format or specifics like pre-releases etc.
- It is a drop-in replacement, i.e. only change is the `import` statement while everything else can be kept which reduces work, see this PR
- Only change is the comparison but that is basically a backported bugfix not a real behavior change, so I wouldn't say it behaves different, just consistent across different python versions (i.e. with and without the bugfix)

If you really want a different name, please suggest one you find better fitting. But note that the diff will be much larger as every use will have to be modified.

Reply:
Yeah, good points, thanks for clarifying! :+1: